survey/views/survey_result.py

```diff
@@ -31,6 +31,8 @@ def serve_result_csv(request, primary_key):
 
     :param int primary_key: The primary key of the survey. """
     survey = get_object_or_404(Survey, pk=primary_key)
+    if not survey.is_published:
+        return HttpResponse(status=302)
```

Comment:
In the test, we expect a 302 when the user is not logged, but it should be a redirection to the login page (in this case probably with an error explaining the problem using messages: https://docs.djangoproject.com/en/2.2/ref/contrib/messages/), not literally an HTTP 302 :)